src/mlpack/methods/lsh_model/lshmodel.hpp

```diff
@@ -0,0 +1,373 @@
+/**
+ * @file lshmodel.hpp
+ * @author Yannis Mentekidis
+ *
+ * Defines the LSHModel class, which models the Locality Sensitive Hashing
+ * algorithm. The model identifies parameter sets that produce satisfactory
+ * results while keeping execution time low.
+ *
+ * The model was proposed by Dong et al in the following paper.
+ *
+ * @code
+ * @article{Dong2008LSHModel,
+ *  author = {Dong, Wei and Wang, Zhe and Josephson, William and Charikar,
+ *      Moses and Li, Kai},
+ *  title = {{Modeling LSH for performance tuning}},
+ *  journal = {Proceeding of the 17th ACM conference on Information and
+ *      knowledge mining - CIKM '08},
+ *  pages = {669},
+ *  url = {http://portal.acm.org/citation.cfm?doid=1458082.1458172},
+ *  year = {2008}
+ * }
+ * @endcode
+ *
+ * We use a different method to fit Gamma Distributions to pairwise distances.
+ * Instead of the MLE method proposed in the paper above, we use the mlpack
+ * class GammaDistribution, which implements fitting according to Thomas Minka's
+ * work.
```

Comment:
I think that Minka's method actually does return the MLE (as your convergence criterion tends to 0), so we are still estimating the MLE, just using a clever algorithm.  Correct me if I'm wrong about that.